beer-garden/functionary
# #144 - Added file parameter support for function detail

Closes #144 

## Introduced Changes
- Added MinIO container to `docker-compose.yml`
- Added `MinioInterface` class that utilizes the MinIO client API to create buckets, put files, delete files, check if files exist, etc.
- Added support for file parameters
- Task file parameters have their values substituted with presigned URLs before being sent to the runner
  - The presigned URLs are never saved to the database. They only need to exist in the task parameters when the task message is sent to the runner
- Added `multipart/form-data` encoding to all forms that spawn tasks
- Containers spawned by the runner now join the functionary network so the container can talk with MinIO

## Testing
- Publish the `file-param` package to Function
- Launch a task from the function detail view for the `Number of Characters` function
- Verify the function launches and displays the number of characters in the file
